CRM/Prospect/Helper/ProspectHelper.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+/**
+ * CRM_Prospect_Helper_ProspectHelper class.
+ */
+class CRM_Prospect_Helper_ProspectHelper {
+
+  const PROSPECT_CASE_TYPE_CATEGORY_NAME = 'Prospecting';
+
+  /**
+   * Checks if the API call belongs to Prospect Category.
+   */
+  public static function isApiCallProspectCategory($case_type_id) {
+    $result = civicrm_api3('CaseType', 'get', [
+      'sequential' => 1,
+      'return' => ['name', 'id'],
+      'case_type_category' => self::PROSPECT_CASE_TYPE_CATEGORY_NAME,
+    ])['values'];
+
+    $prospectCaseTypesByName = \CRM_Utils_Array::rekey($result, 'name');
+    $prospectCaseTypesByID = \CRM_Utils_Array::rekey($result, 'id');
```

Comment:
This function accepts both Case Type ID of Case Type Name. so instead of creating two different functions, can we just rename the variable to $case_type, in the param docs we explain that it can be ID or Name.

that way we dont have to create 2 different functions, also the logic will be more efficient.

And even if we create two different functions, the first function needs to accept ID and a NAME, and then if its not ID, pass it on to another function. so I feel its better to do everything within same function.

like, why make another API call for the same thing.

Reply:
@deb1990 , I agree it is effective but it defeats the purpose of single responsibility, Having a function to accept a parameter that can either be a string or an Integer might not be too good. 
In essence we are saying that this function can Check if it is a prospect category by Case Type Name or check if it is a prospect category by Case Type ID.

I also looked through the code but I did not seem to see a reference as to where we would need to get if a case type is a prospect category by passing the case type name?

IF we need to do this, we can refactor the function into something like this
```php
  public static function isCaseTypeIdProspectCategory($caseTypeId) {
    $params = ['id' => $caseTypeId];
    $result = self::getCaseTypeDetails($params);

    if ($result['count'] > 0) {
      //Means it's a prospect category.
    }
  }

  public static function isCaseTypeNameProspectCategory($caseTypeName) {
    $params = ['name' => $caseTypeName];
    $result = self::getCaseTypeDetails($params);

    if ($result['count'] > 0) {
      //Means it's a prospect category.
    }
  }

  private static function getCaseTypeDetails($params) {
    $defaultParams = [
      'case_type_category' => self::PROSPECT_CASE_TYPE_CATEGORY_NAME,
    ];

    $params = array_merge($defaultParams, $params);
    $result = civicrm_api3('CaseType', 'get', $params)['values'];

    return $result;
  }


```
This will give more clarity as regards the function name and internally both functions are re-using the same logic to fetch the case type details.

What do you think?